presto-geospatial/src/test/java/com/facebook/presto/geospatial/TestGeoQueries.java

```diff
@@ -0,0 +1,346 @@
+        assertQuery("select ST_Y(ST_GeometryFromText('Point (1 2)'))", "select 2.0");
+        assertQueryFails("select ST_Y(ST_GeometryFromText('Polygon ((2 0, 2 1, 3 1))'))", "ST_Y only applies to Point. Input type is: Polygon");
+    }
+
+    @Test
+    public void testSTBoundary()
+            throws Exception
+    {
+        assertQuery("select ST_AsText(ST_Boundary(ST_GeometryFromText('Polygon ((1 1, 4 1, 1 4))')))", "select 'LINESTRING (1 1, 4 1, 1 4, 1 1)'");
+    }
+
+    @Test
+    public void testSTEnvelope()
+            throws Exception
+    {
+        assertQuery("select ST_AsText(ST_Envelope(ST_GeometryFromText('LineString(1 1, 2 2, 1 3)')))", "select 'POLYGON ((1 1, 2 1, 2 3, 1 3, 1 1))'");
+    }
+
+    @Test
+    public void testSTDifference()
+            throws Exception
+    {
+        assertQuery("select ST_AsText(ST_Difference(ST_GeometryFromText('Polygon ((1 1, 1 4, 4 4, 4 1))'), ST_GeometryFromText('Polygon ((2 2, 2 5, 5 5, 5 2))')))", "select 'POLYGON ((1 1, 4 1, 4 2, 2 2, 2 4, 1 4, 1 1))'");
+    }
+
+    @Test
+    public void testSTDistance()
+            throws Exception
+    {
+        assertQuery("select ST_Distance(ST_GeometryFromText('Polygon ((1 1, 1 3, 3 3, 3 1))'), ST_GeometryFromText('Polygon ((4 4, 4 5, 5 5, 5 4))'))", "select 1.4142135623730951");
+    }
+
+    @Test
+    public void testSTExteriorRing()
+            throws Exception
+    {
+        assertQuery("select ST_AsText(ST_ExteriorRing(ST_GeometryFromText('Polygon ((1 1, 1 4, 4 1))')))", "select 'LINESTRING (1 1, 4 1, 1 4, 1 1)'");
+        assertQueryFails("select ST_AsText(ST_ExteriorRing(ST_GeometryFromText('LineString(1 1, 2 2, 1 3)')))", "ST_ExteriorRing only applies to Polygon. Input type is: LineString");
+    }
+
+    @Test
+    public void testSTIntersection()
+            throws Exception
+    {
+        assertQuery("select ST_AsText(ST_Intersection(ST_GeometryFromText('Polygon ((1 1, 1 4, 4 4, 4 1))'), ST_GeometryFromText('Polygon ((2 2, 2 5, 5 5, 5 2))')))", "select 'POLYGON ((2 2, 4 2, 4 4, 2 4, 2 2))'");
+        assertQuery("select ST_AsText(ST_Intersection(ST_GeometryFromText('Polygon ((1 1, 1 4, 4 4, 4 1))'), ST_GeometryFromText('LineString(2 0, 2 3)')))", "select 'LINESTRING (2 1, 2 3)'");
+        assertQuery("select ST_AsText(ST_Intersection(ST_GeometryFromText('POLYGON ((0 0, 0 1, 1 1, 1 0, 0 0))'), ST_GeometryFromText('LINESTRING (-1 1, 1 -1, 1 2)')))", "select 'GEOMETRYCOLLECTION (POINT (-5.9700999704851220e-18 -5.9698882122483090e-18), LINESTRING (1 0, 1 1))'");
+    }
+
+    @Test
+    public void testSTSymmetricDifference()
+            throws Exception
+    {
+        assertQuery("select ST_AsText(ST_SymDifference(ST_GeometryFromText('Polygon ((1 1, 1 4, 4 4, 4 1))'), ST_GeometryFromText('Polygon ((2 2, 2 5, 5 5, 5 2))')))", "select 'MULTIPOLYGON (((1 1, 4 1, 4 2, 2 2, 2 4, 1 4, 1 1)), ((4 2, 5 2, 5 5, 2 5, 2 4, 4 4, 4 2)))'");
+    }
+
+    @Test
+    public void testStContains()
```

Comment:
this method seems to have some duplicate tests; to make it easier to read, consider creating a map between a pair of WKT shapes and a boolean indicating whether shape 1 contains shape 2, then loop over the map to test whether ST_Contains returns expected results; this way it will be easier to see which portion of the test matrix is covered and which is not and whether there are any duplicate tests.